src/xtgeo/io/_file.py

```diff
@@ -578,26 +615,41 @@ def _format_from_contents(self) -> FileFormat:
         # ...
         # The signature here is one float in first line with values 1.0; one string
         # in second line; and 3 or 4 items in the next (sometimes RKB is missing)
-        try:
-            xbuf = buffer.decode().split("\n")
-        except UnicodeDecodeError:
-            return FileFormat.UNKNOWN
+        # Try multiple encodings for cross-platform compatibility (Windows vs macOS)
+        xbuf = None
+        for encoding in ["utf-8", "latin-1", "cp1252"]:
+            try:
+                # Use splitlines() to handle \n (Unix) and \r\n (Windows) endings
+                xbuf = buffer.decode(encoding=encoding).splitlines()
+                break
+            except (UnicodeDecodeError, AttributeError):
+                continue
+
+        if xbuf is None:
+            # If all encodings fail, use latin-1 which accepts all byte values
+            xbuf = buffer.decode(encoding="latin-1", errors="ignore").splitlines()
```

Comment:
I think this change should be reverted. The rest of the code expects utf-8 format (as well as the world, at this point), so allowing to detect the file format with other encodings, while not allowing them to be loaded in the rest of the code if they are not utf-8, doesn't make sense. It is better to fail fast. In particular in FMU, non-utf-8 formats are not supported.

And, 'latin-1' should not be the final fall back. In this case it's better to fail fast.